Dockerfile

```diff
@@ -1,19 +1,85 @@
-# Teeny-tiny matplotlib image based on alpine
-FROM czentye/matplotlib-minimal:3.1.2
+FROM alpine:3.15
+
+# Based on teeny-tiny matplotlib image based on alpine (czentye/matplotlib-minimal:3.1.2)
+# But rebuilding here since alpine:latest is a multi-arch build
 
 LABEL author="Phil Ewels" \
       description="MultiQC" \
       maintainer="phil.ewels@scilifelab.se"
 
-RUN apk add --no-cache bash
+ARG ALPINE_PACKAGE_VERSION="3.15"
+ARG MATPLOTLIB_VERSION="3.1.2"
+
+ARG USER_NAME="multiqc_user"
+ARG USER_ID="1000"
+ARG GROUP_NAME="multiqc_user"
+ARG GROUP_ID="1000"
 
 # Add the MultiQC source files to the container
 ADD . /usr/src/multiqc
 WORKDIR /usr/src/multiqc
 
-# Install MultiQC
-RUN python -m pip install .
+RUN echo "Install deps" 1>&2 && \
+    apk add --repository "https://dl-cdn.alpinelinux.org/alpine/${ALPINE_PACKAGE_VERSION}/main" \
+            --update \
+            --no-cache \
+            bash \
+            build-base \
+            freetype \
+            freetype-dev \
+            libgfortran \
+            libpng \
+            libpng-dev \
+            libstdc++ \
+            python3 \
+            python3-dev \
+            py3-setuptools \
+            py3-pip \
+            py3-wheel && \
+    echo "Link locals and pythons to defaults" 1>&2 && \
+    ln -fs /usr/include/locale.h /usr/include/xlocale.h && \
+    ln -fs /usr/bin/python3 /usr/local/bin/python && \
+    ln -fs /usr/bin/pip3 /usr/local/bin/pip && \
+    echo "Install deps through pip" 1>&2 && \
+    pip3 install -v --no-cache-dir --upgrade \
+      pip && \
+    pip3 install -v --no-cache-dir \
+      numpy && \
+    pip3 install -v --no-cache-dir \
+      matplotlib=="$MATPLOTLIB_VERSION" && \
```

Comment:
Do we need numpy and matplotlib to be installed explicitly here? Can this not be handled by the main MultiQC requirements?

Reply:
> Out of curiosity, does the final image size end up being about the same?

Latest image on ghcr.io for multiqc is 206 Mb, this one is about 266 Mb

> Do we need numpy and matplotlib to be installed explicitly here? Can this not be handled by the main MultiQC requirements?

Tested and looks like we can drop these. I had them in explicitly as I'd thought since you'd set the base image as a specific version as matplotlib, you'd wish to do the same here. 

When matplotlib is just a dependency of multiqc, it appears to compile (which reflects the docs here https://matplotlib.org/stable/users/installing/index.html).  I'm not sure I've seen this when I've specified it on its own pip install line.